recipes/simdjson/all/conanfile.py

```diff
@@ -109,3 +109,5 @@ def package_info(self):
                 self.cpp_info.system_libs.append("pthread")
         if self.options.shared:
             self.cpp_info.defines.append("SIMDJSON_USING_LIBRARY=1")
+            if tools.Version(self.version) >= "0.9.0" and self.settings.compiler == "Visual Studio":
+                self.cpp_info.defines.append("SIMDJSON_USING_WINDOWS_DYNAMIC_LIBRARY=1")
```

Comment:
This block is so weird now: https://github.com/simdjson/simdjson/blob/2db4592571bec79a7e91fee1ad0e0e59217b4fde/include/simdjson/common_defs.h#L150-L191

Reply:
Yes, this will continue to work without new patch.

For pre-0.9.0 using `SIMDJSON_USING_LIBRARY` is valid because the block was much simpler.

For 0.9.0 and later `SIMDJSON_USING_LIBRARY` is guarded by the comment about a specific vcpkg workaround:
```
/**
 * Workaround for the vcpkg package manager. Only vcpkg should
 * ever touch the next line. The SIMDJSON_USING_LIBRARY macro is otherwise unused.
 */
```

I think, using `SIMDJSON_USING_WINDOWS_DYNAMIC_LIBRARY` macro is better for the newer version because it is not tied with `vcpkg` (which can be possibly dropped).

Reply:
yes, I saw, this complexity in `common_defs.h` seems to be useless, I don't know what happened.